common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -78,6 +80,19 @@
   public static final String MERGE_DIR_KEY = "mergeDir";
   public static final String ATTEMPT_ID_KEY = "attemptId";
   private static final int UNDEFINED_ATTEMPT_ID = -1;
+  private static final int UNDEFINED_SHUFFLE_MERGE_ID = Integer.MIN_VALUE;
+
+  // ConcurrentHashMap doesn't allow null for keys or values which is why this is required.
+  // Marker to identify stale shuffle partitions typically happens in the case of
+  // indeterminate stage retries.
+  @VisibleForTesting
+  public static final Map<Integer, AppShufflePartitionInfo> STALE_SHUFFLE_PARTITIONS =
+    new ConcurrentHashMap<>();
```

Comment:
Since we are checking for `STALE_SHUFFLE_PARTITIONS` and `FINALIZED_SHUFFLE_PARTITIONS` separately to respond accordingly if we make both of them `Collections.emptyMap` then that won't work right?

Reply:
Good point, it is the same reference always returned.
In which case, make it an explicit unmodifiable map.
`Collections.unmodifiableMap(new HashMap<>(1))`